### PR TITLE
Fix visible item search regressions[RDY]

### LINKED ISSUE #18925: Regressions in visible item filter
http://smf.cataclysmdda.com/index.php?topic=13515

* Multiple exclusions (for example, `-pulped,-mangled`) don't work
* Double comma (`,,`) locks up the game

### PR DESCRIPTION
**Pull request by epsilon-phase**
Fixes #18925

Namely that negated searches failed to function when combined, curly braces probably broke everything (and they are likely used by users used to the old system), so they're removed from the filter string internally now.

### CODE REVIEW

**Comment by Coolthulhu** [comment]
Doesn't fix `,,` yet - I got a segfault after searching that.

File: src/item_search.cpp [hunk 1]
```diff
@@ -8,34 +8,62 @@
...
+                }
             }
-            auto current_func = item_filter_from_string( current_filter );
-            functions.push_back( current_func );
             filter = trim( filter.substr( comma + 1 ) );
         }
-        if( !filter.empty() ){
-            functions.push_back( item_filter_from_string( filter ) );
+        if( !filter.empty() ) {
+            std::cout << filter << " " << std::endl;
```
Comment (codemime) [inline src/item_search.cpp]: Is that a leftover from debugging?

File: src/item_search.cpp [hunk 2]
```diff
@@ -8,34 +8,62 @@
...
+            }else {
+                inv_functions.push_back( item_filter_from_string( filter) );
+            }
         }
-        return [functions](const item& i) {
-            return std::any_of( functions.begin(), functions.end(),
-            [i](std::function<bool(const item&)> f){
+        return [functions, inv_functions]( const item & i ) {
+            bool p_result = std::any_of( functions.begin(), functions.end(),
+            [i]( std::function<bool( const item & )> f ) {
```
Comment (codemime) [inline src/item_search.cpp]: `Item` should be captured by reference here. Also, I would rename it to `it` (pun unintended).

File: src/item_search.cpp [hunk 3]
```diff
@@ -8,34 +8,62 @@
...
         }
-        if( !filter.empty() ){
-            functions.push_back( item_filter_from_string( filter ) );
+        if( !filter.empty() ) {
+            std::cout << filter << " " << std::endl;
+            if( filter[0] != '-' ) {
+                functions.push_back( item_filter_from_string( filter ) );
+            }else {
+                inv_functions.push_back( item_filter_from_string( filter) );
+            }
```
Comment (codemime) [inline src/item_search.cpp]: [Duplicated](https://github.com/CleverRaven/Cataclysm-DDA/pull/18933/files#diff-2ab5d626787b18b8d3cbabb28d4243d6R32) snippet.

File: src/item_search.cpp [hunk 4]
```diff
@@ -46,9 +74,9 @@ item_filter_from_string(  std::string filter )
             filter = filter.substr( colon + 1 );
         }
     }
-    bool exclude = filter[0] == '-';
+    bool exclude = filter.find( '-' ) != std::string::npos;
```
Comment (codemime) [inline src/item_search.cpp]: Any particular reason for the change?
Actually, you check for negation twice (if exclude duplicates): here and above (when separating functions) which means you do the same work twice: you can either drop [this block](https://github.com/CleverRaven/Cataclysm-DDA/pull/18933/files#diff-2ab5d626787b18b8d3cbabb28d4243d6R80) and `return !f( i );` or leave the functional negation and find a more elegant solution.

**Comment by jaked122** [comment]
You're right. I thought I got all the debug stuff out, guess not.

File: src/item_search.cpp (hunk 1 above)
Comment (jaked122) [inline src/item_search.cpp]: Yes it is.

File: src/item_search.cpp (hunk 4 above)
Comment (jaked122) [inline src/item_search.cpp]: No, this is just less efficient, I'm going to revert that.

**Pull request merged**